src/appleseed/renderer/kernel/shading/shadingpoint.cpp

```diff
@@ -410,6 +410,10 @@ void ShadingPoint::compute_world_space_partial_derivatives() const
             m_dpdu = (dv1 * dp0 - dv0 * dp1) * rcp_det;
             m_dpdv = (du0 * dp1 - du1 * dp0) * rcp_det;
 
+            // Make sure dPdv x dPdu == N
```

Comment:
The comment is not 100% correct, I'll fix it.
The idea is that dPdv x dPdu and dPdy x dPdx point in the same direction as N.
If not, then we flip dPdu or dPdx.